app/services/vacancy_filter.rb

```diff
@@ -0,0 +1,31 @@
+class VacancyFilter
+  attr_reader :managed_school_urns, :managed_organisations
+
+  def initialize(user, organisation)
+    @user_preference = UserPreference.find_or_initialize_by(user_id: user.id, school_group_id: organisation.id)
+
+    @managed_school_urns = @user_preference.managed_school_urns
+    @managed_organisations = @user_preference.managed_organisations
+  end
+
+  def update(params)
+    @managed_school_urns = params[:managed_school_urns]&.reject(&:blank?)
+    @managed_organisations = params[:managed_organisations]
+
+    if managed_organisations.include?('all') || managed_organisations.blank? && managed_school_urns.none?
```

Comment:
Is it more readable and explicit like this? I'm happy either way, but this saves anyone having to think too much about precedence when following the logic
```ruby
if managed_organisations.include?('all') || (managed_organisations.blank? && managed_school_urns.none?)
```